flake8_plugins/assert_expected_actual.py

```diff
@@ -0,0 +1,107 @@
+import ast
+import json
+from pathlib import Path
+import re
+
+
+MSG = 'PTA100 expected should come before actual'
+
+
+def is_literal(node):
+    return (
+        isinstance(node, ast.Constant) or
+        (
+            (
+                isinstance(node, ast.List) or
+                isinstance(node, ast.Set) or
+                isinstance(node, ast.Tuple)
+            ) and
+            all(is_literal(elt) for elt in node.elts)
+        ) or
+        (
+            isinstance(node, ast.Dict) and
+            all(is_literal(key) for key in node.keys) and
+            all(is_literal(value) for value in node.values)
+        )
+    )
+
+
+def classify_arg(node):
+    info = {'type': type(node).__name__, 'literal': is_literal(node)}
+    if info['type'] == 'Name':
+        for keyword in ['actual', 'expected']:
+            if re.search(keyword, node.id, re.IGNORECASE):
+                info[keyword] = True
+    return info
+
+
+def known_pattern(left, right):
+    return right.get('actual') or left.get('expected') or left.get('literal')
+
+
+def classify_call(node):
+    info = {'args': len(node.args)}
+    if info['args'] == 2:
+        left, right = map(classify_arg, node.args)
+        info['left'] = left
+        info['right'] = right
+        if known_pattern(left, right):
+            info['good'] = True
+        # check info.get('good') to ignore cases with two literals
+        if not info.get('good') and known_pattern(right, left):
+            info['flipped'] = True
```

Comment:
not sure if this is too strict - some asserts as comparing for example CPU results with GPU results. so I guess the rule should be 2nd argument should not be literal???

Reply:
could you give an example of an assert that is currently flagged incorrectly? I'm not sure I quite follow